src/main/java/org/snomed/snowstorm/core/data/services/classification/JMSListenerClassificationStatusService.java

```diff
@@ -0,0 +1,53 @@
+package org.snomed.snowstorm.core.data.services.classification;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.activemq.command.ActiveMQTextMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.snomed.snowstorm.core.data.services.classification.pojo.ClassificationStatusResponse;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.jms.annotation.JmsListener;
+import org.springframework.stereotype.Component;
+
+import javax.jms.JMSException;
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+@ConditionalOnProperty(value = "classification-service.job.status.use-jms", havingValue = "true")
+public class JMSListenerClassificationStatusService implements ClassificationStatusService {
+	private static final Logger LOGGER = LoggerFactory.getLogger(JMSListenerClassificationStatusService.class);
+
+	private final ObjectMapper objectMapper = new ObjectMapper();
+	private final Map<String, ClassificationStatusResponse> classificationStatusChanges = new HashMap<>();
```

Comment:
I would use some sort of synchronised map here to avoid the slim possibility that one thread adds something in the same moment as another removing the latest response. Not that slim seeing as there is a thread that pulls statuses out every second.